EIPS/eip-2794.md

```diff
@@ -0,0 +1,121 @@
+
+Hayden Adams. 2018: https://hackmd.io/@477aQ9OrQTCbVR3fq1Qzxg/HJ9jLsfTz?type=view.
+
+[Vitalik Buterin: Improving front running resistance of x*y=k market makers](https://ethresear.ch/t/improving-front-running-resistance-of-x-y-k-market-makers/1281)
+
+The benchmark token NDAO will be forged with ETH based on our proposed RICP algorithm. NDAO's decimals are 18. When the smart contracts that run "forging" start, one will be assigned to NDAO's initial value, ETH's circulating supply will be assigned to ETH's initial total supply and ETH's real-time market cap will be assigned to ETH's initial marketcap .
+
+- Initialized with 1NDAO equaling to 1 USD
+In the RICP algorithm, the value of NDAO is always one ---- this is what we call the benchmark. On the forging contracts' initialization, the NDAO's price in ETH is the same as the USD's price in ETH and the USD's price in ETH is determined by the existing centralized crypto exchanges. This makes NDAO's price an interesting and reasonable start. But right after the forging contracts start, the 1:1 pegging relationship between NDAO and the US dollar will be revoked.
+- On system initialization ETH's circulating supply will be used as the system's initial ETH supply.
+- Technically, on system initialization there are no real NDAOs (only a virtual NDAO supply), and ETH's real-time market cap will be used as the system's initial NDAO supply. Once ETH is sold to this system NDAO will be generated.
+- Virtual trading pairs are established.
+- NDAO will be forged in two ways: realtime forging and staking with forged NDAO
+Realtime forging --- forge NDAOs with ETHs. People who get NDAOs can apply them in various applications.Staking with forged NDAO --- For example, in ETH 2.0 staking, a validator needs to stake 32 ETHs. By using staking with forged NDAO a validator can stake NDAOs instead of ETHs by sending ETHs to the system in exchange for NDAOs as stake. When the validator exits he/she will get back his/her staked NDAOs.
+- The RICP's algorithm runs this way:
+It is initialized with ETH's circulating supply.
+When a trader sends an amount of ETHs to the system, the ETH's circulating supply will be deducted by this amount and the total supply of ETH in the trading pair will be deducted by this amount. We can think of as these ETHs are burned.
+For instance if ETH's total supply is 107,477,373 , a trader sends 100 ETHs to this system in exchange for NDAOs, then the total supply of ETH in the algorithm will be 107,477,273 which will be less than the previous supply by 100, and the ETH's price will rise based on the RICP algorithm.
+- The total supply of ETH in the system and the total supply of NDAO are calculated once on a daily basis.
+- In this system ETH is priced in NDAO.
+
+Note: right after the foring contracts start, NDAO's pegging relationship with USD will be revoked.
+## Improvement to Economic Incentives
```

Comment:
Not a template section.  See https://raw.githubusercontent.com/ethereum/EIPs/master/eip-template.md